lib/src/mapbox_maps_platform.dart

```diff
@@ -181,3 +187,36 @@ class _MapboxMapsPlatform {
     }
   }
 }
+
+/// A registry to hold suffixes for Channels.
+///
+class _SuffixesRegistry {
+  _SuffixesRegistry._instance();
+
+  int _suffix = -1;
+  final Set<int> suffixesInUse = {};
+  final Set<int> suffixesAvailable = {};
+
+  int getSuffix() {
+    int suffix;
+
+    if (suffixesAvailable.isEmpty) {
+      _suffix++;
+      suffixesInUse.add(_suffix);
+      suffix = _suffix;
+    } else {
+      suffix = suffixesAvailable.first;
+      suffixesAvailable.remove(suffix);
+      suffixesInUse.add(suffix);
+    }
```

Comment:
code would be much more readable if Flutter provides collections allowing to do `poll` / `offer` operations

Reply:
Well there are Queue's of course though no `peek/poll` method that gets + removes in one run. Though I've simplified code a bit, thanks for hint.